### PR TITLE
feat(dynamite): Support default responses

### PR DESCRIPTION
**Pull request by provokateurin**
Closes https://github.com/nextcloud/neon/issues/871

I had this laying around from the petstore example. Might as well merge it already now that we have tests for dynamite.

### CODE REVIEW

**Review by Leptopoda — COMMENTED** [review]
How does this handle having both a status code and a default response specified?

I think we really have to make the status code validation taka a function that uses a switch case. This would also allow us to handle status code ranges.

**Comment by provokateurin** [comment]
It is handled the same way as we currently do for multiple status codes: As long as they have the same schema they will be handled the same. If we have a default response then we can ignore if we have other status codes with the same schema because they will be handled by the default already.
I don't think we need your function for this yet (although we certainly need it in the future for supporting multiple schemas). I would like to keep everything as simple as possible until we actually need something more complex.

**Comment by Leptopoda** [comment]
The please also add a test that handles this (having both a default and a status code)

File: packages/dynamite/dynamite_end_to_end_test/lib/responses.openapi.dart [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+  }
+
+  /// This method and the response it returns is experimental. The API might change without a major version bump.
+  ///
+  /// Returns a [Future] containing a [DynamiteRawResponse] with the raw [HttpClientResponse] and serialization helpers.
+  /// Throws a [DynamiteApiException] if the API call does not return an expected status code.
+  ///
+  /// Status codes:
+  ///   * 200
+  ///   * 201
```
Comment (Leptopoda) [inline packages/dynamite/dynamite_end_to_end_test/lib/responses.openapi.dart:103]: This is really bad as the doc comments don't reflect the generated code :grimacing:

**Review by Leptopoda — APPROVED** [review]
LGTM

**Comment by Leptopoda** [comment]
You'll need to rebase as I just merged #1025.

File: packages/dynamite/dynamite_end_to_end_test/lib/responses.openapi.dart (hunk 1 above)
Comment (provokateurin) [reply packages/dynamite/dynamite_end_to_end_test/lib/responses.openapi.dart:103]: Hm true, but it is also good to have this information to handle the other status codes that are not handled by us.

**Pull request merged**